File: 1941-check-if-all-characters-have-equal-number-of-occurrences/1941-check-if-all-characters-have-equal-number-of-occurrences.py
class Solution:
    def areOccurrencesEqual(self, s: str) -> bool:
        count = {}
        for ch in s:
            if ch in count.keys():
                count[ch] += 1
            else:
                count[ch] = 0
        freq = count[s[0]]
        for val in count.values():
            if val != freq:
                return False
        return True

File: 1941-check-if-all-characters-have-equal-number-of-occurrences/1941-check-if-all-characters-have-equal-number-of-occurrences.cpp
class Solution {
public:
    bool areOccurrencesEqual(string s) {
        unordered_map<char, int> um;
        for (auto &ch : s) um[ch]++;
        int freq = um[s[0]];
        for (auto [ch, count] : um) if (count != freq) return false;
        return true;
    }
};